### PR TITLE
Upgrade to rails 7

### PR DESCRIPTION
**Pull request by mileszs**
Upgrade to the latest released version of Rails! Woo hoo!

I did run into a few small issues, particularly with Capybara and Selenium for the systems specs. I abandoned our custom headless chrome in favor of the pre-registered version from Capybara. That seems to work (though I think it is noticeably slower—I don't have any data to back that up, though). 

Otherwise it went fairly smoothly, which makes sense for such a simple Rails application, and one that @t27duck has put quite a bit of work into over the years to clean up. (Thanks, Tony.)

I have tested a bit locally to ensure we didn't break any styles. I have some styling/usability concerns, but they pre-existed this upgrade. Later, I will document them and hopefully resolve them.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -16,7 +16,7 @@ gem 'bootsnap', '>= 1.4.4', require: false
 gem 'jquery-rails'
 gem 'simple_form'
 
-gem 'devise', '4.7.3'
+gem 'devise', '4.8.1'
```
Comment (t27duck) [inline Gemfile:19]: Most recent devise is 4.9.2. Why this (older) version?

File: config/initializers/new_framework_defaults_7_0.rb [hunk 2]
```diff
@@ -0,0 +1,135 @@
+# Be sure to restart your server when you modify this file.
```
Comment (t27duck) [inline config/initializers/new_framework_defaults_7_0.rb:1]: As there is no migration pass for the 7.0 defaults, this whole file can be removed.

File: db/migrate/20230510112953_add_service_name_to_active_storage_blobs.active_storage.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# This migration comes from active_storage (originally 20190112182829)
```
Comment (t27duck) [inline db/migrate/20230510112953_add_service_name_to_active_storage_blobs.active_storage.rb:1]: Why add the active storage tables? (other than "so we can use active storage in the future if we want and the tables are all there ready to go")

File: db/schema.rb [hunk 4]
```diff
@@ -20,28 +19,28 @@
     t.string "encrypted_password", default: "", null: false
     t.integer "failed_attempts", default: 0, null: false
     t.string "unlock_token"
-    t.datetime "locked_at"
-    t.datetime "created_at", null: false
-    t.datetime "updated_at", null: false
+    t.datetime "locked_at", precision: nil
+    t.datetime "created_at", precision: nil, null: false
+    t.datetime "updated_at", precision: nil, null: false
```
Comment (t27duck) [inline db/schema.rb:24]: The churn of these fields is a result in a MySQL fix datetime columns in scheme files. Somewhere in the Rails changelog they explain this.

File: Gemfile (hunk 1 above)
Comment (mileszs) [reply Gemfile:19]: Good point. I just chose the first release that would fix the particular bug I was running into. I'll update it and see what happens.

File: db/schema.rb (hunk 4 above)
Comment (mileszs) [reply db/schema.rb:24]: Ah, cool. This, correct?

image

File: .github/workflows/rubyonrails.yml [hunk 5]
```diff
@@ -40,20 +40,20 @@ jobs:
       - name: Run tests
         run: bin/rake
 
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Install Ruby and gems
+        uses: ruby/setup-ruby@55283cc23133118229fd3f97f9336ee23a179fcf # v1.146.0
```
Comment (t27duck) [inline .github/workflows/rubyonrails.yml:49]: Why this very specific revision of the action?

**Review by t27duck — APPROVED** [review]
- ~~Tests~~ Specs ran
- App boots locally

Probably good enough

File: .github/workflows/rubyonrails.yml (hunk 5 above)
Comment (mileszs) [reply .github/workflows/rubyonrails.yml:49]: From https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-ruby

> GitHub recommends pinning actions to a commit SHA.
> To get a newer version, you will need to update the SHA.
> You can also reference a tag or branch, but the action may change without warning.

The positive view of it is its for your own good / safety. The cynical view is it reduces support requests on GitHub. :-)

**Pull request merged**